My Site Dashboard [Phase 2]: Format Date for Scheduled Posts

ParaskP7 opened the pull request:
Parent: #15200

This PR implements the date excerpt on scheduled posts by formatting the date.

PS: In case you are wondering about Zeplin showing a different date format, this newly date format was discussed and agreed with design, which matches the date format on the posts screen.

-----

Prerequisite:

- Go to `My Site` -> `Me` -> `App Settings` -> `Debug Settings` configuration.
- Turn-on the `MySiteDashboardPhase2FeatureConfig` feature.
- Relaunch the app.

-----

To test:
- If you don't have a scheduled post already, create one.
- Note that on the `My Site` tab, on the `Upcoming scheduled posts` card, any schedule post that is shown within has its date formatted according to the `MMM d` month/day format.
- Also, verify that this date format match the format within the `Posts` screen, and more scecificaly the `SCHEDULED` tab.

-----

Merge instructions:

- [x] Wait for the accompanying [WordPress-FluxC-Android#2207](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2207) PR to be merged.
- [x] Update `fluxcVersion` with the once that includes the above FluxC solution (with either the develop hash or pointing to the newest release). PS: This PR will still show the wrong date until this step is complete.
- [x] Remove the [Status] Not Ready for Merge label.
- [x] Merge this PR.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` sections above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ParaskP7 committed:
Test: Add tests for date excerpt on scheduled posts

These tests will fail as the implementation is not ready yet. The next
commit will make the tests pass (TDD).

ParaskP7 committed:
Feature: Implement date excerpt on scheduled posts

This commit makes the previous failing test pass (TDD).

PS: In case you are wondering about Zeplin showing a different date
format, this newly date format was discussed and agreed with design,
which matches the date format on the posts screen.

ashiagr commented:
Thank you for working on it @ParaskP7! 

> verify that this date format match the format within the Posts screen, and more scecificaly the SCHEDULED tab.

Looks like there's a discrepancy in the scheduled posts display date as we can see in the video:

https://user-images.githubusercontent.com/1405144/144972025-98503cd2-0442-47da-b5e0-bcfb70d0a358.mp4

I selected a time somewhat early in my timezone:

<img width="393" alt="Screenshot 2021-12-07 at 11 02 20 AM" src="https://user-images.githubusercontent.com/1405144/144972304-bf30a7ab-4350-4882-909f-fea6a8953c7f.png">

The scheduled date appeared correctly in my timezone on the `scheduled` posts tab but was set a day earlier on the scheduled post card. 

I see that `post.getDateCreated()` is used on the scheduled post tab which looks a bit confusing to me. We can pair on this and see what's happening.

ParaskP7 commented:
👋 @ashiagr !

> Thank you for working on it @ParaskP7!

Thank you for reviewing it! 🥇 

> Looks like there's a discrepancy in the scheduled posts display date as we can see in the video:
> 
> VIDEO
> 
> I selected a time somewhat early in my timezone:
> 
> SCREENSHOT
> 
> The scheduled date appeared correctly in my timezone on the scheduled posts tab but was set a day earlier on the scheduled post card.
> 
> I see that post.getDateCreated() is used on the scheduled post tab which looks a bit confusing to me. We can pair on this and see what's happening.

Thanks so much for capturing all that.

Question (❓): I am not sure if you have tested this PR with its accompanying [WordPress-FluxC-Android#2207](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2207) PR (see merge instructions), but I think that what you are seeing is going to be solved but it. Can you please check it and verify that? 🙏 

PS: This PR is only for formatting purposes, the PR that is actually solves the timezone discrepancies are within FluxC.

ashiagr commented:
> PS: This PR is only for formatting purposes, the PR that is actually solves the timezone discrepancies are within FluxC.

Oh, I see. I just followed the prerequisites. I'll test it out now.

ParaskP7 commented:
> Oh, I see. I just followed the prerequisites. I'll test it out now.

Many thanks Ashita and apologies for any confusion that I caused! 🙇

ashiagr reviewed (approved):
Approved ✅ 

Works well with the accompanying FluxC PR. 🙌 

Feel free to update `FluxC` version and merge it thereafter.

ParaskP7 commented:
👋 @ashiagr !

> Works well with the accompanying FluxC PR. 🙌

Great, many thanks for cross checking! 🎉 

> Feel free to update FluxC version and merge it thereafter.

Will do, many thanks again! 💯

ParaskP7 committed:
Build: Update fluxc hash

This 'FluxC' hash updates 'FluxC' to that 'develop' version where
the API endpoint date for the 'post' cards get converted to GMT
timezone.
